.github/workflows/tests.yml

```diff
@@ -0,0 +1,42 @@
+on:
+  workflow_call:
+name: Tests
+jobs:
+  run:
+    strategy:
+      matrix:
+        go-version: [1.16.x, 1.17.x]
+        os: [ubuntu-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go-version }}
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.6
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Install Python dependencies
+        run: |
+          python -m pip install --upgrade iso8601 requests securesystemslib six tuf
+      - name: Format Unix
+        if: runner.os == 'Linux'
+        run: test -z $(go fmt ./...)
+      - name: Test
+        run: go test -race -covermode atomic -coverprofile='profile.cov' ./...
+      - name: Send coverage
+        if: runner.os == 'Linux'
+        env:
+          COVERALLS_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          GO111MODULE=off go get github.com/mattn/goveralls
+          $(go env GOPATH)/bin/goveralls -coverprofile=profile.cov -service=github
+      - name: Vet
+        run: go vet ./...
+      - name: Install staticcheck
+        run: "go install honnef.co/go/tools/cmd/staticcheck@v0.2.2"
```

Comment:
Wonder if it would make sense to use golangci-lint instead of staticcheck. It comes with staticchecks but a ton of useful other linters (including security related linters)

Reply:
@shibumi - Actually, I was planning to add that in another PR as I was afraid it was going to introduce a lot of additional changes to this one, but now that it was suggested - I tried it and it turned out to be okay. I kept the same linters we had so far and added a few others. At some point, we can enable more linters in a separate PR along with fixes for the issues they raise.